File: cpp.rx/sum.cpp
#include <iostream>

#include "rxcpp/rx.hpp"

int main(int argc, char *argv[])
{
  auto v = rxcpp::observable<>::range(1, 10).sum();

  v.subscribe(
    [](int v) { std::cout << v << std::endl; },
    []{}
  );

  return 0;
}

File: go/sum.go
package main

import "fmt"

func sum(a *[3] float64) (sum float64) {
  for _, v := range *a {
    sum += v
  }

  return
}

func main() {
  arr := [...] float64 {1.0, 2., 3.}
	fmt.Println(sum(&arr))
}
